cmd/sovereignnode/incomingHeader/incomingSCRs.go

```diff
@@ -0,0 +1,82 @@
+package incomingHeader
+
+import (
+	"encoding/hex"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+const (
+	minTopicsInEvent  = 4
+	numTransferTopics = 3
+)
+
+type scrInfo struct {
+	scr  *smartContractResult.SmartContractResult
+	hash []byte
+}
+
+func (ihp *incomingHeaderProcessor) createIncomingSCRs(events []data.EventHandler) ([]*scrInfo, error) {
+	scrs := make([]*scrInfo, 0, len(events))
+
+	for _, event := range events {
+		topics := event.GetTopics()
+		if len(topics) < minTopicsInEvent || len(topics[1:])%numTransferTopics != 0 {
+			log.Error("incomingHeaderHandler.createIncomingSCRs",
+				"error", errInvalidNumTopicsIncomingEvent,
+				"num topics", len(topics),
+				"topics", topics)
+			continue
+		}
+
+		scr := &smartContractResult.SmartContractResult{
+			RcvAddr: topics[0],
+			SndAddr: core.ESDTSCAddress,
+			Data:    createSCRData(topics),
+		}
+
+		hash, err := core.CalculateHash(ihp.marshaller, ihp.hasher, scr)
+		if err != nil {
+			return nil, err
+		}
+
+		scrs = append(scrs, &scrInfo{
+			scr:  scr,
+			hash: hash,
+		})
+	}
+
+	return scrs, nil
+}
+
+func createSCRData(topics [][]byte) []byte {
+	numTokensToTransfer := len(topics[1:]) / numTransferTopics
+	numTokensToTransferBytes := big.NewInt(int64(numTokensToTransfer)).Bytes()
+
+	ret := []byte(core.BuiltInFunctionMultiESDTNFTTransfer +
+		"@" + hex.EncodeToString(topics[0]) + // topics[0] = address
+		"@" + hex.EncodeToString(numTokensToTransferBytes))
+
+	for idx := 1; idx < len(topics[1:]); idx += 3 {
+		transfer := []byte("@" +
+			hex.EncodeToString(topics[idx]) + // tokenID
+			"@" + hex.EncodeToString(topics[idx+1]) + //nonce
+			"@" + hex.EncodeToString(topics[idx+2])) //value
+
+		ret = append(ret, transfer...)
+	}
+
+	return ret
+}
+
+func (ihp *incomingHeaderProcessor) addSCRsToPool(scrs []*scrInfo) {
+	cacheID := process.ShardCacherIdentifier(core.MainChainShardId, core.SovereignChainShardId)
```

Comment:
this introduces an overhead by having a SCR pool with the Metachain and there is no metachain on the Sovereign. Could the "intrashard" pool be used?

Reply:
That's not **meta**chain, it is actually **main**chain.
I don't think it should cause overheads, since it is exactly what we want to achieve: simulate a "sharded" pool from main->sovereign.
Also, from my understanding, the code should be able to process/interpret "cross chain" scrs/mbs this way (cc @SebastianMarian for confirmation)

Also (2), from what I've looked through the code, when we create a mb with :
- senderShard = mainChain
- recevierShard = sovereignChain
processor should look at the mb sender/receiver ids and take the tx hashes from the sharded tx pool

Reply:
ok! since it is already implemented like this, it should be ok